pipestat/backend.py

```diff
@@ -0,0 +1,1226 @@
+import sys
+import os
+from abc import ABC
+from glob import glob
+from logging import getLogger
+from yacman import YAMLConfigManager
+from ubiquerg import create_lock, remove_lock, expandpath
+from contextlib import contextmanager
+
+from sqlalchemy import text
+from sqlmodel import Session, select as sql_select
+
+from .const import *
+from .exceptions import *
+from .helpers import *
+
+if int(sys.version.split(".")[1]) < 9:
```

Comment:
Why are we checking version before import typing?

Reply:
This is related to type hinting and backwards compatibility with Python 3.8.
[PEP585](https://peps.python.org/pep-0585/#implementation)